raiden/network/proxies/token.py

```diff
@@ -163,12 +170,27 @@ def balance_of(
             block_identifier=block_identifier
         )
 
-    def total_supply(self, block_identifier: BlockSpecification = "latest") -> TokenAmount:
-        """ Return the total supply of the token at the given block identifier. """
-        return TokenAmount(
-            self.proxy.contract.functions.totalSupply().call(block_identifier=block_identifier)
+    def total_supply(
+        self, block_identifier: BlockSpecification = "latest"
+    ) -> Optional[TokenAmount]:
+        """ Return the total supply of the token at the given block identifier.
+
+        Because Token is just an interface, it is not possible to check the
+        bytecode during the proxy instantiation. This means it is possible for
+        the proxy to be instantiated with a a smart contrat address of the
+        wrong type (a non ERC20 contract), or a partial implementation of the
+        ERC20 standard (the funciton totalSupply is missing). If that happens
```

Comment:
s/funciton/function

Besides: Did you test the behavior if the property doesn't exist. In the explorer we had some problems with this and it's solved as: https://github.com/raiden-network/explorer/blob/master/backend/metrics_backend/utils/token.py#L30